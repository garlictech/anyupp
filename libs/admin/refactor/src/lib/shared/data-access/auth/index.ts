import { CognitoService } from './services/cognito/cognito.service';
import { AuthGuard } from './services/auth-guard/auth-guard';

export { CognitoService, AuthGuard };
