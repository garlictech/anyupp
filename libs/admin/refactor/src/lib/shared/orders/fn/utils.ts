import { CrudSdk } from '@bgap/crud-gql/api';

export interface OrderHandlerDeps {
  crudSdk: CrudSdk;
  timestamp: () => number;
}
