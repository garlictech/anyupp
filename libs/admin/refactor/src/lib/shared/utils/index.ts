import { ImageCompressorService } from './services/image-compressor/image-compressor.service';
import { QrGeneratorService } from './services/qr-generator/qr-generator.service';
import { ToasterService } from './services/toaster/toaster.service';

export * from './const';
export * from './enums';
export * from './fn';
export { ImageCompressorService, ToasterService, QrGeneratorService };
