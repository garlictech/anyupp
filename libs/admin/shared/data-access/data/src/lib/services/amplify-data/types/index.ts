import {
  GetAdminUserQuery,
  GetChainProductQuery,
  GetChainQuery,
  GetGroupQuery,
  GetOrderQuery,
  GetProductCategoryQuery,
  GetUnitQuery,
  GetUserQuery,
  ListAdminUsersQuery,
  ListChainProductsQuery,
  ListChainsQuery,
  ListGroupsQuery,
  ListOrdersQuery,
  ListProductCategorysQuery,
  ListUnitsQuery,
  ListUsersQuery,
  OnAdminUserChangeSubscription,
  OnChainsChangeSubscription,
  OnGroupsChangeSubscription,
  OnProductCategoriesChangeSubscription,
  OnUnitsChangeSubscription,
  OnUsersChangeSubscription,
  OnChainProductChangeSubscription
} from '@bgap/admin/amplify-api';

export type queryTypes = GetAdminUserQuery &
  GetChainQuery &
  GetGroupQuery &
  GetUnitQuery &
  GetOrderQuery &
  GetProductCategoryQuery &
  GetChainProductQuery &
  GetUserQuery &
  GetChainProductQuery;
export type listTypes = ListAdminUsersQuery &
  ListChainsQuery &
  ListGroupsQuery &
  ListUnitsQuery &
  ListOrdersQuery &
  ListProductCategorysQuery &
  ListChainProductsQuery &
  ListUsersQuery &
  ListChainProductsQuery;
export type apiQueryTypes = queryTypes & listTypes;
export type subscriptionTypes = OnAdminUserChangeSubscription &
  OnChainsChangeSubscription &
  OnGroupsChangeSubscription &
  OnUnitsChangeSubscription &
  OnUsersChangeSubscription &
  OnProductCategoriesChangeSubscription &
  OnChainProductChangeSubscription;
