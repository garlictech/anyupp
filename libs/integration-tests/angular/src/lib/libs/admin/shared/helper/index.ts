import { Auth } from 'aws-amplify';

import { awsConfig } from '@bgap/crud-gql/api';
import {
  getCognitoUsername,
  testAdminUserPassword,
} from '@bgap/shared/fixtures';

export const signInToCognito = async () => {
  Auth.configure({
    ...awsConfig,
    authenticationFlowType: 'USER_PASSWORD_AUTH',
    aws_appsync_authenticationType: 'AMAZON_COGNITO_USER_POOLS',
  });

  await Auth.signIn(getCognitoUsername('monad'), testAdminUserPassword);
};

export const signOutFromCognito = async () => {
  await Auth.signOut();
};
