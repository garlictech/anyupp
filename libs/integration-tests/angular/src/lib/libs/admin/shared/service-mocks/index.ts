export class MockNbDialogService {
  open() {
    /** */
  }
}

export class MockTranslateService {
  get() {
    /** */
  }
}
