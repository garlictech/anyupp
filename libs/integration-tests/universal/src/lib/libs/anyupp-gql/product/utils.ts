import * as CrudApi from '@bgap/crud-gql/api';
import {
  validateChainProduct,
  validateGroupProduct,
  validateUnitProduct,
} from '@bgap/shared/data-validators';
import { throwError } from 'rxjs';
import { catchError, switchMap } from 'rxjs/operators';

export const getChainProduct = (sdk: CrudApi.CrudSdk, productId: string) => {
  return sdk.GetChainProduct({ id: productId }).pipe(
    switchMap(validateChainProduct),
    catchError(err => {
      console.error(err);
      return throwError('ChainProdcuct is missing');
    }),
  );
};

export const getGroupProduct = (sdk: CrudApi.CrudSdk, productId: string) => {
  return sdk.GetGroupProduct({ id: productId }).pipe(
    switchMap(validateGroupProduct),
    catchError(err => {
      console.error(err);
      return throwError('GroupProdcuct is missing');
    }),
  );
};

export const getUnitProduct = (sdk: CrudApi.CrudSdk, productId: string) => {
  return sdk.GetUnitProduct({ id: productId }).pipe(
    switchMap(validateUnitProduct),
    catchError(err => {
      console.error(err);
      return throwError('UnitProdcuct is missing');
    }),
  );
};
