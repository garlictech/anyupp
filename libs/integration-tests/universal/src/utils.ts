import * as R from 'ramda';
import { pipe } from 'fp-ts/lib/function';

export const ES_DELAY = 3000; //ms
export const DYNAMO_DELAY = 1000; //ms

// Replace UUID-s with a static string for jest matchers
export const maskV4UuidIds = <T>(data: T) =>
  pipe(
    data,
    JSON.stringify,
    R.replace(
      /[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/g,
      'MASKED ID',
    ),
    (d: string) => JSON.parse(d) as T,
  );
