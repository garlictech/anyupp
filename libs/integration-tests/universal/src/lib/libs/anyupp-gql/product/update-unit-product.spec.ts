import { Auth } from '@aws-amplify/auth';
import * as CrudApi from '@bgap/crud-gql/api';
import { AnyuppSdk } from '@bgap/anyupp-gql/api';
import { productRequestHandler } from '@bgap/anyupp-gql/backend';
import * as CrudApi from '@bgap/crud-gql/api';
import {
  productFixture,
  testAdminUsername,
  testAdminUserPassword,
  testIdPrefix,
} from '@bgap/shared/fixtures';
import { RequiredId } from '@bgap/shared/types';
import { defer, iif, of, throwError } from 'rxjs';
import { catchError, delay, switchMap, takeLast } from 'rxjs/operators';
import {
  createAuthenticatedAnyuppSdk,
  createIamCrudSdk,
} from '../../../../api-clients';
import {
  createTestUnitProduct,
  deleteTestUnitProduct,
} from '../../../seeds/unit-product';
import { getUnitProduct } from './utils';

const DYNAMODB_OPERATION_DELAY = 3000;
const TEST_NAME = 'UPDATE_UNIT_PROD_';
const DEBUG_MODE_TEST_WITH_LOCALE_CODE = false;

const unitProduct: RequiredId<CrudApi.CreateUnitProductInput> = {
  ...productFixture.unitProductInputBase,
  id: `${testIdPrefix}${TEST_NAME}unitProduct_01`,
  unitId: 'UNIT_ID_TO_CRUD',
};
describe('UpdateUnitProduct tests', () => {
  let authAnyuppSdk: AnyuppSdk;
  const iamCrudSdk: CrudApi.CrudSdk = createIamCrudSdk();

  const cleanup = deleteTestUnitProduct(unitProduct.id, iamCrudSdk);

  beforeAll(async () => {
    authAnyuppSdk = await createAuthenticatedAnyuppSdk(
      testAdminUsername,
      testAdminUserPassword,
    )
      .toPromise()
      .then(x => x.authAnyuppSdk);

    await cleanup
      .pipe(
        takeLast(1),
        switchMap(() => createTestUnitProduct(unitProduct, iamCrudSdk)),
        takeLast(1),
        catchError(err => {
          console.error('BEFORE HOOK ERROR');
          return throwError(err);
        }),
      )
      .toPromise();
  });

  afterAll(async () => {
    await cleanup.toPromise();
    await Auth.signOut();
  });

  it('should throw in case of an invalid input', done => {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    const input: any = { foo: 'NOT A VALID INPUT' };
    iif(
      () => DEBUG_MODE_TEST_WITH_LOCALE_CODE,
      defer(() =>
        productRequestHandler({
          crudSdk: iamCrudSdk,
          regenerateUnitDataHandler: jest.fn(),
        }).updateUnitProduct({ input }),
      ),
      authAnyuppSdk.UpdateUnitProduct({ input }),
    ).subscribe({
      next() {
        console.error(`${TEST_NAME}Test ERROR`, 'SHOULD NOT SUCCEED');
      },
      error(err) {
        expect(err).toMatchSnapshot();
        done();
      },
    });
  }, 15000);

  it('should throw because of the missing GroupProduct - ANYUPP_API CALL', done => {
    const input: CrudApi.UpdateUnitProductInput = {
      id: unitProduct.id,
      position: 10,
    };

    authAnyuppSdk
      .UpdateUnitProduct({ input })
      .pipe(
        delay(DYNAMODB_OPERATION_DELAY),
        switchMap(product => getUnitProduct(iamCrudSdk, product.id)),
      )
      .subscribe({
        next() {
          console.error(`${TEST_NAME}Test ERROR`, 'SHOULD NOT SUCCEED');
        },
        error(err) {
          expect(err).toMatchSnapshot();
          done();
        },
      });
  }, 25000);

  it('should update the unitProduct in the database and call the regeneration logic - LOCAL test', done => {
    const regenMockHandler = jest.fn().mockReturnValue(of(true));
    const input: CrudApi.UpdateUnitProductInput = {
      id: unitProduct.id,
      position: 10,
    };

    defer(() =>
      productRequestHandler({
        crudSdk: iamCrudSdk,
        regenerateUnitDataHandler: regenMockHandler,
      }).updateUnitProduct({
        input,
      }),
    )
      .pipe(
        delay(DYNAMODB_OPERATION_DELAY),
        switchMap(product => getUnitProduct(iamCrudSdk, product.id)),
      )
      .subscribe({
        next(result) {
          const { createdAt, updatedAt, ...product } = result;
          expect(createdAt).not.toBeUndefined();
          expect(updatedAt).not.toBeUndefined();
          expect(product).toMatchSnapshot('POSITION SHOULD BE 10');

          // Regeneration check
          expect(regenMockHandler).toBeCalled();
          expect(regenMockHandler).toHaveBeenNthCalledWith(
            1,
            unitProduct.unitId,
          );
          done();
        },
      });
  }, 25000);
});
