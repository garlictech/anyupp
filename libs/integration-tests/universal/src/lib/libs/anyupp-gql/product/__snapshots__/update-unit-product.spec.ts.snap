// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateUnitProduct tests should throw because of the missing GroupProduct - ANYUPP_API CALL 1`] = `[Error: GraphQL error: GroupProduct Object Validation Error (JOI): "value" must be of type object]`;

exports[`UpdateUnitProduct tests should throw in case of an invalid input 1`] = `[Error: GraphQL error: The variables input contains a field name 'foo' that is not defined for input object type 'UpdateUnitProductInput' ]`;

exports[`UpdateUnitProduct tests should update the unitProduct in the database and call the regeneration logic - LOCAL test: POSITION SHOULD BE 10 1`] = `
Object {
  "chainId": "chainId_",
  "configSets": null,
  "groupId": "groupId_",
  "id": "test_UPDATE_UNIT_PROD_unitProduct_01",
  "isVisible": true,
  "laneId": "laneId_",
  "parentId": "parentId_",
  "position": 10,
  "takeaway": true,
  "unitId": "UNIT_ID_TO_CRUD",
  "variants": Array [
    Object {
      "availabilities": Array [
        Object {
          "dayFrom": "",
          "dayTo": "",
          "price": 1.5,
          "timeFrom": "",
          "timeTo": "",
          "type": "A",
        },
      ],
      "id": "test_UnitProductVariant_id_1",
      "isAvailable": true,
      "pack": Object {
        "size": 1,
        "unit": "UNIT_UNIT",
      },
      "position": 1,
      "price": 30,
      "refGroupPrice": 30,
      "variantName": Object {
        "de": null,
        "en": "VARIANT_NAME_1",
        "hu": null,
      },
    },
    Object {
      "availabilities": Array [
        Object {
          "dayFrom": "",
          "dayTo": "",
          "price": 3,
          "timeFrom": "",
          "timeTo": "",
          "type": "A",
        },
      ],
      "id": "test_UnitProductVariant_id_2",
      "isAvailable": true,
      "pack": Object {
        "size": 2,
        "unit": "UNIT_UNIT",
      },
      "position": 2,
      "price": 60,
      "refGroupPrice": 60,
      "variantName": Object {
        "de": null,
        "en": "VARIANT_NAME_2",
        "hu": null,
      },
    },
  ],
}
`;
